src/core/reader/types/values.rs

```diff
@@ -40,6 +40,22 @@ impl WasmReader<'_> {
         Ok(result)
     }
 
+    pub fn read_var_f64(&mut self) -> Result<u64> {
+        if self.full_wasm_binary.len() - self.pc < 8 {
+            return Err(Error::Eof);
+        }
+
+        let word = u64::from_le_bytes(
+            self.full_wasm_binary[self.pc..self.pc + 8]
+                .try_into()
+                .unwrap(),
+        );
+
+        self.strip_bytes::<8>()?;
+
+        Ok(word)
+    }
+
```

Comment:
Would this be possible here instead?

```suggestion
        let bytes = self.strip_bytes::<8>().map_err(|_| Error::Eof)?;
        let word = u64::from_le_bytes(bytes);
```

Reply:
This should also be possible:
```rs
let bytes = self.strip_bytes::<{ std::mem::size_of::<u64>() }>().map_err(|_| Error::Eof)?;
```

Reply:
Used first suggestion